kvbc/test/sparse_merkle_storage/sparse_merkle_db_editor_test.cpp

```diff
@@ -333,46 +350,56 @@ TEST_F(SparseMerkleDbInspectorTests, get_value_with_version_0) {
-            run(CommandLineArguments{{"sparse_merkle_db_inspector_test", rocksDbPath(), "getValue"}}, out_, err_));
+            run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "getValue"}}, out_, err_));
   ASSERT_TRUE(out_.str().empty());
   ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: Missing HEX-KEY argument"));
 }
 
 TEST_F(SparseMerkleDbInspectorTests, get_value_invalid_key_size) {
   ASSERT_EQ(
       EXIT_FAILURE,
-      run(CommandLineArguments{{"sparse_merkle_db_inspector_test", rocksDbPath(), "getValue", "0xabc"}}, out_, err_));
+      run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "getValue", "0xabc"}}, out_, err_));
   ASSERT_TRUE(out_.str().empty());
   ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: Invalid hex string: 0xabc"));
 }
 
 TEST_F(SparseMerkleDbInspectorTests, get_value_invalid_key_chars_at_start) {
-  ASSERT_EQ(
-      EXIT_FAILURE,
-      run(CommandLineArguments{{"sparse_merkle_db_inspector_test", rocksDbPath(), "getValue", "0xjj"}}, out_, err_));
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "getValue", "0xjj"}}, out_, err_));
   ASSERT_TRUE(out_.str().empty());
   ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: Invalid hex string: 0xjj"));
 }
 
 TEST_F(SparseMerkleDbInspectorTests, get_value_invalid_key_chars_at_end) {
   ASSERT_EQ(
       EXIT_FAILURE,
-      run(CommandLineArguments{{"sparse_merkle_db_inspector_test", rocksDbPath(), "getValue", "0xabcx"}}, out_, err_));
+      run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "getValue", "0xabcx"}}, out_, err_));
   ASSERT_TRUE(out_.str().empty());
   ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: Invalid hex string: 0xabcx"));
 }
 
 TEST_F(SparseMerkleDbInspectorTests, get_value_invalid_key_chars_no_0x) {
-  ASSERT_EQ(
-      EXIT_FAILURE,
-      run(CommandLineArguments{{"sparse_merkle_db_inspector_test", rocksDbPath(), "getValue", "abck"}}, out_, err_));
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "getValue", "abck"}}, out_, err_));
   ASSERT_TRUE(out_.str().empty());
   ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: Invalid hex string: abck"));
 }
 
+TEST_F(SparseMerkleDbInspectorTests, remove_metadata) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(CommandLineArguments{{"sparse_merkle_db_editor_test", rocksDbPath(), "removeMetadata"}}, out_, err_));
+  ASSERT_THAT(out_.str(), StartsWith(toJson(std::string{"result"}, std::string{"true"})));
+  ASSERT_TRUE(err_.str().empty());
+
+  const auto db = TestRocksDb::create();
+  const auto& kvp = generateMetadataAndStateTransfer();
+  for (const auto& kv : kvp) {
+    ASSERT_TRUE(db->has(kv.first).isNotFound());
+  }
```

Comment:
When we call `TestRocksDb::create()`, we will essentially create a new DB handle **and** call `cleanup()`, leading to always having an empty DB.

I think what we can do instead is to make the DB object a member of the fixture(Test) and just use it to verify keys are deleted after the `removeMedatada` call. Or any other method of making sure we use the same DB on disk.

Another thing I think might make sense is to verify that other key types are present after calling `removeMetadata`, i.e. `Block`, `Key` and so on.

Reply:
I did it first but then I had 2 instances of the DB object(fixture and the code under test).
As a result, the code under test could not open the DB(the `LOCK` file issue).
As far as I know, `SetUp` and `TearDown` are called for each test, aren't they?

I've removed `cleanup()` from `create` and call explicitly in `SetUp`.
Thus, both the editor and the test use the same DB.
Added checks to make sure that blocks still exist.

Reply:
I think you are right, calling `cleanup()` in `create()` might not be the best idea.

Thanks!